src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,362 @@
+It is a dekadal product and is updated around 1 week to
+10 days after the end of the period.
+
+While the calculated cumulative biomasse and anomaly data
+is presented at a yearly level on the GeoSahel portal, it
+is not available in dekadal form. The only dekadal form data
+is for DMP values downsampled to the ADMIN0-2 levels across
+the region of interested. The code below accesses this dekadal
+data and then recalculates cumulative biomasse and anomaly
+using that. Since DMP measurements are productivity values,
+measuring the daily dry matter productivity across the dekad of
+interest, the calculation of total biomasse in a year is taken
+by averaging the DMP over the entire period and multiplying by
+365.25. The default year starts in the 10th dekad (1st of
+April) until the last dekad of March, to correspond with the
+West African monsoon season.
+"""
+
+import os
+from pathlib import Path
+from typing import List, Literal, Union, get_args
+
+import numpy as np
+import pandas as pd
+import rioxarray
+
+from src.utils_general.utils import download_ftp
+
+_BASE_URL = (
+    "http://80.69.76.253:8080/geoserver"
+    "/Biomass/wfs?&REQUEST="
+    "GetFeature&SERVICE=wfs&VERSION=1.1.0"
+    "&TYPENAME=WA_DMP_{admin}_ef_v0&"
+    "outputformat=CSV&srsName=EPSG:4326"
+)
+
+_RAW_FILENAME = "WA_DMP_{admin}_ef_v0.csv"
+
+_RAW_PATH = Path(os.getenv("AA_DATA_DIR"), "public", "raw", "glb", "biomasse")
+
+
+def _get_raw_path(admin):
+    return Path(_RAW_PATH, _RAW_FILENAME.format(admin=admin))
+
+
+_PROCESSED_FILENAME = "biomasse_{iso3}_{admin}_dekad_{start_dekad}.csv"
+
+
+def _get_processed_path(admin, start_dekad, iso3: Union[None, str] = None):
+    if iso3 is None:
+        iso3 = "glb"
+
+    _processed_path = Path(
+        os.getenv("AA_DATA_DIR"), "public", "processed", iso3, "biomasse"
+    )
+
+    return Path(
+        _processed_path,
+        _PROCESSED_FILENAME.format(
+            iso3=iso3, admin=admin, start_dekad=start_dekad
+        ),
+    )
+
+
+AdminArgument = Literal["ADM0", "ADM1", "ADM2"]
+_VALID_ADMIN = get_args(AdminArgument)
+
+
+def _check_admin(admin):
+    if admin not in _VALID_ADMIN:
+        raise ValueError("`admin` must be one of 'ADM0', 'ADM1', or 'ADM2'.")
+
+
+def download_dmp(admin: AdminArgument = "ADM2"):
+    """Download raw DMP data
+
+    Raw DMP data is downloaded for specified
+    administrative area. This data is downloaded
+    for all countries in West Africa covered by
+    Biomasse, using those specific countries'
+    administrative boundaries, and is downloaded
+    for all available years. The data is downloaded
+    in CSV format and can be processed using
+    ``process_dmp()``.
+    """
+    _check_admin(admin)
+    url = _BASE_URL.format(admin=admin)
+    raw_path = _get_raw_path(admin)
+    download_ftp(url=url, save_path=raw_path)
+    return None
+
+
+def load_dmp(admin: AdminArgument = "ADM2", redownload: bool = False):
+    _check_admin(admin)
+    raw_path = _get_raw_path(admin)
```

Comment:
hmm don't you want to check here for file existence instead of redownloading every time you load the dataset?

Reply:
Was all a bit of a mixup switching to `unlink`, woops!